tests/2d.js

```diff
@@ -562,106 +624,105 @@
       evt.x = 1; // bad boy!
     });
-    e.one("MotionChange", function(evt) { strictEqual(evt.key, "vy"); strictEqual(evt.oldValue, -1); });
-    e.vy = 0;
-    Crafty.timer.simulateFrames(1);
-
-    e.vx = 0;
-    e.one("MotionChange", function(evt) { strictEqual(evt.key, "vy"); strictEqual(evt.oldValue, 0); });
-    e.vy = 1;
-    e.one("MotionChange", function(evt) { strictEqual(evt.key, "vy"); strictEqual(evt.oldValue, 1); });
     e.vy = 0;
     Crafty.timer.simulateFrames(1);
 
 
+    
+    // Group 4, rotation tests
     e.vrotation = 0;
-    Crafty.timer.simulateFrames(1);
-
-    e.one("MotionChange", function(evt) { strictEqual(evt.key, "vrotation"); strictEqual(evt.oldValue, 0); });
+    e.one("MotionChange", function(evt) {
+      strictEqual(evt.key, "vrotation", "[4] - change of angular speed");
+      strictEqual(evt.oldValue, 0, "[4] - old vr value was 0"); 
+    });
     e.vrotation = 1;
-    e.one("MotionChange", function(evt) { strictEqual(evt.key, "vrotation"); strictEqual(evt.oldValue, 1); });
+    e.one("MotionChange", function(evt) {
+      strictEqual(evt.key, "vrotation", "[4] - change of angular speed");
+      strictEqual(evt.oldValue, 1, "[4] - old value was 1");
+    });
     e.vrotation = 0;
-    Crafty.timer.simulateFrames(1);
 
-    e.one("NewRevolution", function(evt) {
-      equal(evt, 1);
+    // Group 5
+    e.one("NewRotationDirection", function(evt) {
+      equal(evt, 1, "[5] - Rotating in the positive sense");
+    });
+    e.one("Rotated", function(oldValue) { 
+      strictEqual(oldValue, 0, "[5] - Rotated from 0 position");
     });
-    e.one("Rotated", function(oldValue) { strictEqual(oldValue, 0); });
-    e.one("MotionChange", function(evt) { strictEqual(evt.key, "vrotation"); strictEqual(evt.oldValue, 0); });
     e.vrotation = 1;
     Crafty.timer.simulateFrames(1);
 
-    e.one("NewRevolution", function(evt) {
-      equal(evt, -1);
+    var old_rotation = e.rotation;
+    e.one("NewRotationDirection", function(evt) {
+      equal(evt, -1, "[6] - Rotating in the negative sense");
+    });
+    e.one("Rotated", function(oldValue) { 
+      strictEqual(oldValue, old_rotation, "[6] - Old rotation matches cached value");
     });
-    e.one("Rotated", function(oldValue) { strictEqual(oldValue, 1); });
-    e.one("MotionChange", function(evt) { strictEqual(evt.key, "vrotation"); strictEqual(evt.oldValue, 1); });
     e.vrotation = -1;
     Crafty.timer.simulateFrames(1);
 
-    e.one("NewRevolution", function(evt) {
-      equal(evt, 0);
+    e.one("NewRotationDirection", function(evt) {
+      equal(evt, 0, "[7] - No longer rotating");
     });
-    e.one("MotionChange", function(evt) { strictEqual(evt.key, "vrotation"); strictEqual(evt.oldValue, -1); });
     e.vrotation = 0;
     Crafty.timer.simulateFrames(1);
 
-    equal(newDirectionEvents, 4);
-    equal(newRevolutionEvents, 3);
-    equal(movedEvents, 6);
-    equal(rotatedEvents, 2);
-    equal(motionEvents, 17);
```

Comment:
this checks shouldn't be removed, as we can not guarantee that all event callbacks fired as we wanted